src/main/java/apoc/schema/Schemas.java

```diff
@@ -117,6 +135,101 @@ private AssertSchemaResult createUniqueConstraint(Schema schema, String lbl, Str
         return new AssertSchemaResult(lbl, key).unique().created();
     }
 
+    public List<AssertSchemaResult> recreateIndexes(long allIndexesTimeout, long dropAllIndexesTimeout, long recreateAllIndexesTimeout) throws ExecutionException, InterruptedException, IllegalArgumentException, TimeoutException {
+        Schema schema = db.schema();
+        List<AssertSchemaResult> result = Collections.synchronizedList(new ArrayList<>());
+        ExecutorService executorService = Pools.SINGLE;
+        List<IndexDefinition> indexesToRecreate = Util.inTxFuture(executorService, db, () -> {
+            List<IndexDefinition> idx = new ArrayList<>();
+            Iterable<IndexDefinition> outerIndexes = schema.getIndexes();
+            for (Iterator<IndexDefinition> it = outerIndexes.iterator(); it.hasNext(); ) {
+                IndexDefinition index = it.next();
+                if (!index.isConstraintIndex()) {
+                    idx.add(index);
+                }
+            }
+            return idx;
+        }).get(allIndexesTimeout, TimeUnit.MILLISECONDS);
+
+        Util.inTxFuture(executorService, db, () -> {
+            for (IndexDefinition definition : indexesToRecreate) {
+                String label = definition.getLabel().name();
+                List<String> keys = new ArrayList<>();
+                definition.getPropertyKeys().forEach(keys::add);
+
+                AssertSchemaResult info = new AssertSchemaResult(label, keys);
```

Comment:
should also contain the type of index/constraint

Reply:
and compound vs. single

Reply:
This class is copied from the existing `apoc.schema.assert`procedure, we haven't done anything different here